weatherTask/weatherman.py

```diff
@@ -1,151 +1,146 @@
-                        'date': datetime.strptime(temp[0], '%Y-%m-%d').strftime('%B %d'),
-                        'temp_max': int(temp[1]),
-                        'temp_min': int(temp[3]),
-                        'humidity': int(temp[7])
-                    })
-
-def monthly_weather_data(year, month, folderpath):
-    """Function to read monthly data"""
-    month_name = calendar.month_abbr[int(month)]
-    file_path = os.path.join(folderpath, f"lahore_weather_{year}_{month_name}.txt")
-    if not os.path.exists(file_path):
-        return None
-    monthly_data = []
-    with open(file_path, encoding="utf-8") as file:
-        lines = file.readlines()[2:-1]
-        for line in lines:
-            temp = line.split(",")
-            if month and temp[0].split("-")[1].lower() != month:
-                continue
-            if temp[1] and temp[3] and temp[7]:
-                monthly_data.append({
-                    'date': datetime.strptime(temp[0], '%Y-%m-%d').strftime('%d'),
-                    'temp_max': int(temp[1]),
-                    'temp_min': int(temp[3]),
-                    'humidity': int(temp[7])
-                })
-    return monthly_data
-
-def highest_lowest_temperature(d):
-    """Function to find highest and lowest temperature data"""
-    if not d:
-        return None
-
-    highest_temp = d[0]
-    lowest_temp = d[0]
-
-    for i in d:
-        if i['temp_max'] > highest_temp['temp_max']:
-            highest_temp = i
-        if i['temp_min'] < lowest_temp['temp_min']:
-            lowest_temp = i
-
-    return highest_temp, lowest_temp
-
-def highest_humidity(d):
-    """Function to get highest humidity data"""
-    if not d:
-        return None
-
-    high_humidity = d[0]
-
-    for i in d:
-        if i['humidity'] > high_humidity['humidity']:
-            high_humidity = i
-
-    return high_humidity
-
-def average_data(d):
-    """Function to get average temperature and humidity data"""
-    if not d:
-        return None
-
-    temp_max = 0
-    temp_min = 0
-    humidity = 0
-
-    for i in d:
-        temp_max += i['temp_max']
-        temp_min += i['temp_min']
-        humidity += i['humidity']
-
-    data_points = len(d)
-    avg_temp_max = temp_max / data_points
-    avg_temp_min = temp_min / data_points
-    avg_humidity = humidity / data_points
-
-    return avg_temp_max, avg_temp_min, avg_humidity
-
-def draw_bar_chart(chart_data):
-    """Function to get chart data"""
-    for d in chart_data:
-        print(f"{d['date']} {colored('+' * d['temp_max'], 'red')} {d['temp_max']}C")
-        print(f"{d['date']} {colored('+' * d['temp_min'], 'blue')} {d['temp_min']}C")
-
-def draw_singleline_bar_chart(chart_data):
-    """Function to get singleline chart data"""
-    for d in chart_data:
-        print(f"{d['date']} {colored('+' * d['temp_min'], 'blue')}"
-        f"{colored('+' * d['temp_max'], 'red')} {d['temp_min']}C - {d['temp_max']}C")
+            lines = file.readlines()[1:-1]
```

Comment:
I guess this is done to handle empty rows, lets make it more generic to handle empty row anywhere, indexing is not considered a good practice